src/middlewared/middlewared/api/v25_04_0/virt_instance.py

```diff
@@ -98,6 +100,8 @@ def validate_attrs(self):
         else:
             if self.enable_vnc and self.vnc_port is None:
                 raise ValueError('VNC port must be set when VNC is enabled')
+            if self.vnc_password and self.vnc_password.get_secret_value() and not self.enable_vnc:
```

Comment:
Why do you need `if self.vnc_password and self.vnc_password.get_secret_value()` here? If it's provided, you just need to do `if self.vnc_password is not None and self.enable_vnc`